moving_modules.md

```diff
@@ -0,0 +1,224 @@
+
+## Module Updates
+
+As an admin, if there is a problem with a type/provider packaged with puppet-agent, I want to be able to install a newer version from the forge (for both serverless and server-based), so that I don't need to wait for Puppet to release a new puppet-agent build.
+
+ ## Puppet-Agent Updates
+
+As an admin, when the puppet-agent package is updated, I want to be able to use the new versions of types and providers that come with the new package, even if that means overwriting and deleting older preinstalled versions. However, installing a new puppet-agent package should not clobber modules I've installed via puppet module tool, r10k, etc.
+
+## Module Pinning
+
+As an admin, if I update the puppet-agent package, and it updates a preinstalled module, but the module introduces a regression, I want to easily install the older version of the module from the forge.  I don't want to be forced to downgrade puppet-agent packages, because that process introduces more risk.
+
+# File Path Changes
+
+Puppet's default `basemodulepath` includes two module directories visible to all environments:
+
+    /etc/puppetlabs/code/modules
+    /opt/puppetlabs/puppet/modules
+
+The first is where modules are typically installed to via puppet module tool, r10k, codemanager/filesync. The second path typically contains modules that are installed in PE environments, though there is nothing stopping users from manually installing modules there, e.g. `puppet module tool install puppetlabs-apt --target-dir /opt/puppetlabs/puppet/modules`. It is important that preinstalled modules do not use those same locations, otherwise, it will confuse package managers.
+
+We propose a new directory `/opt/puppetlabs/puppet/vendored_modules` to be created at puppet-agent installation time and containing all modules added to puppet-agent at packaging. The directory should be appended to the default `basemodulepath` so that the modules are available during compilation and application.
+
+# Resource Type Taxonomy
+
+## Internal
+
+The following types are internal to Puppet and will remain as is:
+
+    component
+    schedule
+    stage
+    whit
+
+## Core
+
+The following types will be left in Puppet for now. The `file`, `user`, and `group` types are needed to apply settings catalogs. The `filebucket`, `resources`, and `tidy` types know too much about puppet internals to be removed. The `notify` type is used extensively in puppet rspec tests as it is the most basic (providerless) type. The `package` and `service` types have multiple providers for each type, which makes removal more difficult. We may remove them at a later time (TBD).
```

Comment:
i suggest using the same language as the intro - "extract" instead of "remove", so it doesn't feel like they are being *deleted*, just pulled into a different place.